src/persist-txn/src/lib.rs

```diff
@@ -314,6 +324,14 @@ pub trait TxnsCodec: Debug {
     ///
     /// Implementations should panic if the values are invalid.
     fn decode(key: Self::Key, val: Self::Val) -> TxnsEntry;
+
+    /// Returns if a part might include the given data shard based on pushdown
+    /// stats.
+    ///
+    /// False positives are okay (needless fetches) but false negatives are not
+    /// (incorrectness). Returns an Option to make `?` convenient, `None` is
+    /// treated the same as `Some(true)`.
```

Comment:
It could be because it's Friday afternoon, but I'm having a little trouble wrapping my head around the return type. 

- `None` means that you SHOULD fetch the part.
- `Some(true)` also means that you SHOULD fetch the part.
- `Some(false)` means that you SHOULD NOT fetch the part.

The `None` case is the opposite of what I would have expected.

Reply:
Yeah, it might be clearer if we flipped the semantics of the method to be `should_filter_part`, in which case we get:

- `None` I have no idea (so default to fetch)
- `Some(false)` I am sure we should fetch
- `Some(true)` I am sure we should filter

I had this as the current `should_fetch_part` semantics to match the ones in `persist_source`, @bkirwi how concerned would you be about the two not matching? (Should we also flip the persist_source one? I admit, I have to carefully think about this every time and maybe `should_fetch_part` would make it easier to reason about `persist_source` too.)

Reply:
Not very concerned, but a light preference for flipping them both if you think it may help there too!

Reply:
Sgtm, but flipping the meaning of the bool in shard_source/persist_source is a scary enough change that I want it to be its own PR, so I'll do both of these as a followup

Reply:
Okay this is up in https://github.com/MaterializeInc/materialize/pull/22974

> flipping the meaning of the bool in shard_source/persist_source is a scary enough change that I want it to be its own PR

This was prescient, I had at least two bugs in the first version of this change